### PR TITLE
[stable] Slightly re-extend the C++ interface for LDC

### PR DESCRIPTION
**Pull request by kinke**
After @ibuclaw's cut-down in v2.089.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Auto-merge toggled on

**Comment by kinke** [comment]
@thewilsonator: Please let Iain have a look at it, he might be against exposing all `Identifier::<foo>` globals again.

**Comment by thewilsonator** [comment]
Auto-merge toggled off

File: src/dmd/root/outbuffer.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ struct OutBuffer
     void reserve(size_t nbytes);
     void setsize(size_t size);
     void reset();
-    //void write(const void *data, d_size_t nbytes);
+    void write(const void *data, size_t nbytes);
```
Comment (ibuclaw) [inline src/dmd/root/outbuffer.h]: This should be d_size_t due to OSX.

**Comment by ibuclaw** [comment]
> @thewilsonator: Please let Iain have a look at it, he might be against exposing all Identifier::<foo> globals again.

It's not much of an optimization IMO, and you can always just cache the pointers yourself if you really need to.

That said, if you're giving them C++ linkage, you should probably update id.h to reflect what is exposed/available to the code generator. :-)

File: src/dmd/root/outbuffer.h (hunk 1 above)
Comment (kinke) [reply src/dmd/root/outbuffer.h]: All other methods here use `size_t`... only DMD 2.079-2.081 IIRC are affected by the temporary size_t interop regression; LDC has never suffered from it.

**Comment by kinke** [comment]
> you should probably update id.h to reflect what is exposed/available to the code generator

Well that is destined to be outdated right away, nobody is going to keep that in sync with the autogenerated globals. For LDC, we currently only declare 69 globals we actually need.

File: src/dmd/root/outbuffer.h (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/root/outbuffer.h]: It's not there only for the size_t regression, there are some platforms where you will suffer from it. ;-)

Comment (ibuclaw) [reply src/dmd/root/outbuffer.h]: Regardless, I'll add a test for this in the cxx-unittester, and I'll be forced to change it to d_size_t anyway...

Comment (kinke) [reply src/dmd/root/outbuffer.h]: > there are some platforms where you will suffer from it

32-bit macOS isn't supported any longer, and all others use `d_size_t = size_t` as of now.

> I'll be forced to change it to d_size_t

Alright then. I still count about 45 `size_t` occurrences in the C++ headers, and most of those are function params.

**Comment by ZombineDev** [comment]
Perhaps it's a good idea to add a call to these functions to https://github.com/dlang/dmd/blob/master/src/tests/cxxfrontend.c, in order to ensure someone else doesn't remove them from the C++ API again.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**